fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateCTEConsumeRule.java

```diff
@@ -0,0 +1,44 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.sql.optimizer.rule.transformation;
+
+import com.google.common.collect.Lists;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptimizerContext;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.logical.LogicalCTEConsumeOperator;
+import com.starrocks.sql.optimizer.operator.logical.LogicalFilterOperator;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.rule.RuleType;
+
+import java.util.List;
+
+/*
+ *          Filter          CTEConsume(Predicate)
+ *            |                 |
+ *        CTEConsume   =>     Filter
+ *            |                 |
+ *           Node              Node
+ *
+ * */
+public class PushDownPredicateCTEConsumeRule extends TransformationRule {
```

Comment:
Whether need PushDownLimitCTEConsumeRule？

Reply:
Yes, you can find it in PushDownLimitDirectRule